src/ServiceControl/CustomChecks/SqlDbCustomCheckDataStore.cs

```diff
@@ -0,0 +1,212 @@
+                    new { Id = id }).ConfigureAwait(false);
+
+                if (customCheck == null ||
+                    ((int)customCheck.Status == (int)Status.Fail && !detail.HasFailed) ||
+                    ((int)customCheck.Status == (int)Status.Pass && detail.HasFailed))
+                {
+                    publish = true;
+                }
+
+                await connection.ExecuteAsync(
+                    @"IF EXISTS(SELECT * FROM [CustomChecks] WHERE Id = @Id)
+                            UPDATE [CustomChecks] SET
+                                [CustomCheckId] = @CustomCheckId,
+                                [Category] = @Category,
+                                [Status] = @Status,
+                                [ReportedAt] = @ReportedAt,
+                                [FailureReason] = @FailureReason,
+                                [OriginatingEndpoint] = @OriginatingEndpoint
+                            WHERE [Id] = @Id
+                          ELSE
+                            INSERT INTO [CustomChecks](Id, CustomCheckId, Category, Status, ReportedAt, FailureReason, OriginatingEndpoint) 
+                            VALUES(@Id, @CustomCheckId, @Category, @Status, @ReportedAt, @FailureReason, @OriginatingEndpoint)",
+                    new
+                    {
+                        Id = id,
+                        detail.CustomCheckId,
+                        detail.Category,
+                        detail.ReportedAt,
+                        detail.FailureReason,
+                        Status = detail.HasFailed ? Status.Fail : Status.Pass,
+                        OriginatingEndpoint = mapper.Serialize(detail.OriginatingEndpoint)
+                    }).ConfigureAwait(false);
+            }
+
+            if (publish)
+            {
+                if (detail.HasFailed)
+                {
+                    await domainEvents.Raise(new CustomCheckFailed
+                    {
+                        Id = id,
+                        CustomCheckId = detail.CustomCheckId,
+                        Category = detail.Category,
+                        FailedAt = detail.ReportedAt,
+                        FailureReason = detail.FailureReason,
+                        OriginatingEndpoint = detail.OriginatingEndpoint
+                    }).ConfigureAwait(false);
+                }
+                else
+                {
+                    await domainEvents.Raise(new CustomCheckSucceeded
+                    {
+                        Id = id,
+                        CustomCheckId = detail.CustomCheckId,
+                        Category = detail.Category,
+                        SucceededAt = detail.ReportedAt,
+                        OriginatingEndpoint = detail.OriginatingEndpoint
+                    }).ConfigureAwait(false);
+                }
+            }
+        }
+
+        public async Task<StatisticsResult> GetStats(HttpRequestMessage request, string status = null)
```

Comment:
Would it be possible to push the `PagingInfo` abstraction to this call (and ideally to the interface)? That would prevent the Http abstraction from pushing down this far.

Reply:
I have changed the interface so it doesn't rely on `HttpRequestMessage` anymore.